---
title: "Correlation of HNRNPH and PSI levels in BRCA tumor samples"
author:
- name: Mario Keller
  affiliation: Faculty of Biological Sciences, Goethe University Frankfurt
output:
    BiocStyle::html_document:
      toc: TRUE
      toc_float: TRUE
      code_folding: hide
editor_options: 
  markdown: 
    wrap: 72
---

```{r}
knitr::opts_chunk$set(echo = TRUE, message=FALSE, warning=FALSE,
                      crop=NULL, results = TRUE)
```

```{r libraries}
library(tidyverse)
library(DESeq2)
library(knitr)
library(ggbeeswarm)
library(kableExtra)
library(ggrastr)
library(ggpointdensity)
library(viridis)
library(ggpubr)
```

```{r defineTheme}
myTheme <- theme_bw() +
    theme(axis.text = element_text(size = 14, colour="black"),
          axis.title = element_text(size=16, colour="black"),
          axis.ticks=element_line(color="black"),
          axis.ticks.length=unit(.15, "cm"),
          panel.border=element_rect(color="black", fill = NA),
          panel.background = element_blank(),
          plot.background = element_blank(),
          legend.text = element_text(size=12),
          legend.position = "none")
```

```{r paths}
projectDir <- "/Users/jb/data/HNPRNPH/Tretow_et_al_2023"

tcgaDir <- paste0(projectDir, "/8_TCGA_analysis")
```

# Background

The data so far led to the identification of CE and ALE events showing
cooperatively enhanced and repressed HNRNPH regulation. As the RNA-seq
experiments were performed in the human breast cancer cell line MCF7, we
raised the question whether these events show a similar dependence on
HNRNPH levels and direction of regulation in the TCGA BRCA cohort. To
test this, expression of *HNRNPH1* and *HNRNPH2* and PSI levels of the
enhanced and repressed CE and ALE events were computed for each BRCA
tumor sample. In addition, the non-regulated CE events were also checked to
see if these are also not affected in the BRCA samples.

# Data

Vst-transformed and batch corrected expression values are loaded from a
RDS-File created with *Calculate_expression_BRCA.R*.

Sample meta information is loaded from a RDS-File created with
*Calculate_expression_BRCA.R*.

PSI values of junctions of interest are loaded from a RDS-File created
with *Calculate_PSIs_BRCA.R*.

```{r}

vsdBatchCorrected<- readRDS(paste0(tcgaDir,"/rds_files/BRCA_vsdBatchCorrected.rds"))

metaInformation <- readRDS(paste0(tcgaDir,"/rds_files/BRCA_metaInformation.rds"))

PSIs <- readRDS(paste0(tcgaDir,"/rds_files/BRCA_PSI.rds"))
```

A data.frame with the expression levels of *HNRNPH1* (ENSG00000169045.17)
and *HNRNPH2* (ENSG00000126945.8) is created.

```{r}
# Each row is a sample and columns are the samplebarcode and HNRNPH1 and HNRNPH2
#   expression levels
HNRNPHexpression <- data.frame(geneName=c("HNRNPH1", "HNRNPH2"),
                               vsdBatchCorrected[match(c("ENSG00000169045.17",
                                                         "ENSG00000126945.8"),
                                                       rownames(vsdBatchCorrected)),] %>%
                                   assay) %>%
    pivot_longer(., cols=-1, names_to="tcgaSampleBarcode", values_to="vst") %>%
    mutate(tcgaSampleBarcode=gsub(".", "-", tcgaSampleBarcode, fixed = T)) %>% 
    pivot_wider(., names_from = geneName, values_from=vst)

# The participant barcode and the sample type are added 
HNRNPHexpression <- HNRNPHexpression %>% 
    left_join(., metaInformation %>% dplyr::select(tcga.tcga_barcode,
                                                   tcga.gdc_cases.submitter_id,
                                                   tcga.gdc_cases.samples.sample_type),
              by=c("tcgaSampleBarcode" = "tcga.tcga_barcode")) %>%
    dplyr::rename(tcgaPatientBarcode = tcga.gdc_cases.submitter_id,
           sampleType = tcga.gdc_cases.samples.sample_type)
```

Splicing events are reduced to those of the category "Coop-Enh",
"Coop-Rep" and "nonReg". Note that Coop-Enh and Coop-Rep comprise CE and
ALE events.

```{r}

# Remove the quantifications od the HNRNPH1 NMD Isoform and 
#   use for the non-regulated CE events "nonReg" as Hill category
PSIs <- PSIs %>% 
    dplyr::filter(eventType != "HNRNPH1nmdIsoform") %>%
    dplyr::mutate(hillCat = ifelse(is.na(hillCat), "nonReg", hillCat))

# Filter for the three Hill categories of interest
PSIs <- PSIs %>% dplyr::filter(hillCat %in% c("Coop-Enh", "Coop-Rep", "nonReg"))
```

# HNRNPH1 and HNRNPH2 levels in normal and tumor samples

To check if there is a difference in HNRNPH1/2 levels between normal and tumor
samples, I plot the respective expression levels for the 108 patients with
available normal and tumor quantifications.

```{r}
HNRNPHexpression %>% group_by(tcgaPatientBarcode) %>% filter(n() ==2) %>% ungroup %>% arrange(tcgaPatientBarcode, sampleType) %>%
    pivot_longer(., cols=c("HNRNPH1", "HNRNPH2"), names_to="gene", values_to="vst") %>%
    mutate(., sampleType = factor(sampleType, levels=c("Solid Tissue Normal", "Primary Tumor"), labels = c("Normal", "Tumor"))) %>%
    ggplot(., aes(x=sampleType, y=vst, fill=sampleType)) +
    geom_boxplot(width=.2, alpha=1, position = position_nudge(x = c(-.15, .15)), outlier.size = -1) +
    geom_point(pch=21, size=3, alpha=0.5) +
    geom_path(mapping=aes(group=tcgaPatientBarcode), size=1, alpha=0.5) +
    scale_y_continuous(breaks=1:15) +
    labs(x = "Sample type", y="Expression (vst)") +
    stat_compare_means(paired=TRUE) +
    facet_wrap(~gene, ncol=2, scales = "free_y") +
    myTheme +
    theme(aspect.ratio = 2/1)
```

# Correlation HNRNPH1 and HNRNPH2

Next, the correlation of HNRNPH1 and HNRNPH2 levels was checked for tumor samples.

```{r}

ggplot(HNRNPHexpression %>% dplyr::filter(sampleType == "Primary Tumor"), aes(x=HNRNPH1, y=HNRNPH2)) +
    rasterise(geom_pointdensity(size=3), dpi=300) +
    geom_smooth(method="lm", se=FALSE) +
    scale_x_continuous(breaks=1:16) +
    scale_y_continuous(breaks=1:16) +
    scale_color_viridis() +
    stat_cor() +
    labs(x="HNRNPH1 expression (vst)", y="HNRNPH2 expression (vst)") +
    myTheme +
    theme(aspect.ratio=1)
```

# Correlations between *HNRNPH1/2* and Enhanced, Repressed and Ctrl events

The PSI values cooperatively enhanced and repressed CE and ALE events as
well as the non-regulated CE events are correlated with the expression
of *HNRNPH1* and *HNRNPH2*.

Correlations were performed for an event if at least 10 tumor samples had a
PSI value for the event. Correlations were considered significant if
their adjusted P value (BH correction) was $\le$ 0.05. Boxplots are are
based on significant and non-signififcant correlations. Significance is
shown by the transparency (alpha) of the points.

```{r}

corRes <- lapply(PSIs$eventID %>% unique, function(tmpEventID){
    
    # Combine HNRNPH expression levels and PSI values in a single data.frame
    df <- inner_join(HNRNPHexpression,
                         PSIs %>%
                             dplyr::filter(eventID == tmpEventID) %>%
                             dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI),
                     by = c("tcgaSampleBarcode", "tcgaPatientBarcode", "sampleType")) %>%
        dplyr::filter(sampleType=="Primary Tumor")
        
    # If there are less than 10 samples return an empty data.frame
    if(nrow(df) < 10){
        return(data.frame())
    }
    
    # Create the output data.frame
    df <- data.frame(gene=c("HNRNPH1", "HNRNPH2"),
                     eventID=tmpEventID,
                     hillCat= PSIs %>% dplyr::filter(eventID == tmpEventID) %>%
                         pull(hillCat) %>% unique,
                     r=c(cor.test(df$HNRNPH1,
                                  df$PSI)$estimate,
                         cor.test(df$HNRNPH2,
                                  df$PSI)$estimate),
                     pval=c(cor.test(df$HNRNPH1,
                                     df$PSI)$p.value,
                            cor.test(df$HNRNPH2,
                                     df$PSI)$p.value))
    return(df)
}) %>% bind_rows()

# Do the FDR corecction
corRes <- corRes %>% mutate(pvalAdj = p.adjust(pval, method = "BH")) %>%
    mutate(sign=ifelse(pvalAdj <= 0.05, TRUE, FALSE))

corRes$hillCat <- factor(corRes$hillCat, levels=c("Coop-Enh","nonReg", "Coop-Rep"), labels=c("Enhanced", "Ctrl", "Repressed"))

ggplot(mapping=aes(x=gene, y=r, col=hillCat, fill=hillCat, alpha=sign)) +
        geom_quasirandom(data=corRes %>% dplyr::filter(sign), dodge.width = .8, size=1.25) +
        geom_quasirandom(data=corRes %>% dplyr::filter(!sign), dodge.width = .8, size=1.25) +
        geom_boxplot(data=corRes, alpha=.5, outlier.size = -1, col="black",
                     position = position_dodge(width = .8), width = .7) +
        scale_fill_manual(values=c("Enhanced"="cornflowerblue", "Repressed"="salmon2", "Ctrl"="#606060")) +
        scale_color_manual(values=c("Enhanced"="cornflowerblue", "Repressed"="salmon2", "Ctrl"="#606060")) +
        scale_alpha_manual(values=c("TRUE" = 1, "FALSE" = .2)) +
        coord_cartesian(ylim=c(-1, 1)) +
        scale_y_continuous(breaks=seq(-1, 1, by=0.5)) +
        labs(x="", y="Correlation coefficnt") +
        myTheme + theme(legend.position = "right", aspect.ratio=1)

```

# New JONAS
```{r}
test_res <- rbind((aov(r ~ hillCat, data = corRes %>% filter(gene=="HNRNPH1")) %>% rstatix::tukey_hsd()) %>% cbind(gene="HNRNPH1"), 
                  (aov(r ~ hillCat, data = corRes %>% filter(gene=="HNRNPH2")) %>% rstatix::tukey_hsd()) %>% cbind(gene="HNRNPH2")) #%>% rstatix::add_x_position(x="gene", dodge=0.8)
test_res$y.position <- seq(0.65, 1000, by=0.1) %>% head(nrow(test_res)/2) %>% rep(2)
test_res$xmin <- c(0.73, 0.73, 1, 1.73, 1.73, 2)
test_res$xmax <- c(1, 1.27, 1.27, 2, 2.27, 2.27)

ggplot(data=corRes, mapping=aes(x=gene, y=r)) +
        geom_quasirandom(data=corRes, aes(col=hillCat, fill=hillCat), dodge.width = .8, size=1.25) +
        geom_boxplot(data=corRes, aes(col=hillCat, fill=hillCat), alpha=.5, outlier.size = -1, col="black", position = position_dodge(width = .8), width = .7) +
        ggpubr::stat_pvalue_manual(test_res, tip.length=0) +
        scale_fill_manual(values=c("Enhanced"="cornflowerblue", "Repressed"="salmon2", "Ctrl"="#606060")) +
        scale_color_manual(values=c("Enhanced"="cornflowerblue", "Repressed"="salmon2", "Ctrl"="#606060")) +
        coord_cartesian(ylim=c(-0.6, 0.9)) +
        scale_y_continuous(breaks=seq(-1, 1, by=0.5)) +
        labs(x="", y="Correlation coefficnt") +
        myTheme + theme(legend.position = "right", aspect.ratio=1.5)
```

At least for *HNRNPH2* we can see the expected regulation, namely that
enhanced CE and ALE events are positively correlated with *HNRNPH2* levels
and repressed CE and ALE events negatively. The non-regulated events
have a median correlation coefficient of around 0.

# Correlation examples

## IL17RC - Repressed

```{r}
id <- "ENSG00000163702.20_7_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=1) +
        labs(x="HNRNPH1 expression (vst)", y="IL17RC exon 18 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=1) +
        labs(x="HNRNPH2 expression (vst)", y="IL17RC exon 18 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

## NFE2L1 - Enhanced

```{r}
id <- "ENSG00000082641.16_4_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,.8)) +
        scale_color_viridis() +
        stat_cor(label.y=.8) +
        labs(x="HNRNPH1 expression (vst)", y="NFE2L1 exon 5 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,.8)) +
        scale_color_viridis() +
        stat_cor(label.y=.8) +
        labs(x="HNRNPH2 expression (vst)", y="NFE2L1 exon 5 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

## MYL6 - Repressed

```{r}
id <- "ENSG00000092841.19_2_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=1) +
        labs(x="HNRNPH1 expression (vst)", y="MYL6 exon 6 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=1) +
        labs(x="HNRNPH2 expression (vst)", y="MYL6 exon 6 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

## PIP4K2C - Enhanced

```{r}
id <- "ENSG00000166908.18_1_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0.6,1)) +
        scale_color_viridis() +
        stat_cor(label.y=.6) +
        labs(x="HNRNPH1 expression (vst)", y="PIP4K2C exon 5 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0.6,1)) +
        scale_color_viridis() +
        stat_cor(label.y=.6) +
        labs(x="HNRNPH2 expression (vst)", y="PIP4K2C exon 5 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

## MST1R - Repressed

```{r}
id <- "ENSG00000164078.14_1_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=0) +
        labs(x="HNRNPH1 expression (vst)", y="MST1R exon 11 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0,1)) +
        scale_color_viridis() +
        stat_cor(label.y=0) +
        labs(x="HNRNPH2 expression (vst)", y="MST1R exon 11 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

## AKR1A1 - Enhanced

```{r}
id <- "ENSG00000117448.14_3_CE_1"

plotDF <- inner_join(HNRNPHexpression,
                     PSIs %>% dplyr::filter(eventID == id) %>% dplyr::select(tcgaPatientBarcode, tcgaSampleBarcode, sampleType, PSI)
                     ) %>% dplyr::filter(sampleType=="Primary Tumor") 

ggplot(plotDF, aes(x=HNRNPH1, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0.4,1)) +
        scale_color_viridis() +
        stat_cor(label.y=0.4) +
        labs(x="HNRNPH1 expression (vst)", y="AKR1A1 exon 7 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)

ggplot(plotDF, aes(x=HNRNPH2, y=PSI)) +
        rasterise(geom_pointdensity(size=3), dpi=300) +
        geom_smooth(method="lm", se=FALSE) +
        scale_x_continuous(breaks=1:16) +
        coord_cartesian(ylim=c(0.4,1)) +
        scale_color_viridis() +
        stat_cor(label.y=0.4) +
        labs(x="HNRNPH2 expression (vst)", y="AKR1A1 exon 7 (PSI)") +
        myTheme +
        theme(aspect.ratio = 1)
```

# Session Information

```{r}
sessionInfo()
```
